pkg/common-controller/groupsnapshot_controller_helper.go

```diff
@@ -649,6 +673,29 @@ func (ctrl *csiSnapshotCommonController) updateGroupSnapshotStatus(groupSnapshot
 		if !utils.IsGroupSnapshotReady(groupSnapshotObj) && utils.IsGroupSnapshotReady(groupSnapshotClone) {
 			msg := fmt.Sprintf("GroupSnapshot %s is ready to use.", utils.GroupSnapshotKey(groupSnapshot))
 			ctrl.eventRecorder.Event(groupSnapshot, v1.EventTypeNormal, "GroupSnapshotReady", msg)
+
+			// The VolumeGroupSnapshot resource is now ready to use. There's no need for the
+			// groupsnapshot.storage.kubernetes.io/info annotation anymore so we can remove it.
+			if metav1.HasAnnotation(groupSnapshotContent.ObjectMeta, utils.AnnSnapshotInfo) {
```

Comment:
It appears that you've only removed AnnSnapshotInfo from here, but it is still added to the VolumeGroupSnapshotContent.